### PR TITLE
Update installation guide to match how-to template

### PR DESCRIPTION
**Pull request by DragaDoncila**
This PR updates the installation guide to match our new docs template:
- brings prerequisites to the top
- highlights that we strongly recommend installing as a package
- adds admonitions in relevant spots to make it easier to browse
- adds dropdowns for the three different package install options to make it easier to browse
- tidies up some wording
- consolidates links and suggestions in the "next steps" section

### CODE REVIEW

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md:8]: I think we may want to keep this redirection for people looking for the developer installation guide instead

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]: Can we flag this as the method as "recommended"?

The bundle is still a bit experimental, and it's not so easy for people to add new packages to the bundle environment, etc. I think generally we prefer if people use napari installed as a python package.

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]: Possibly we should talk about the restrictions involved with choosing the bundled app over the other methods

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]: What's this line about? Was there supposed to be bold text here?

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [reply tutorials/fundamentals/installation.md:8]: Ah, that link doesn't work anyway. Here it is: https://napari.org/developers/contributing.html

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]: I don't think this content should live here - editable pip installations are for the developer docs (and these instructions are also slightly different to what's written there https://napari.org/developers/contributing.html)

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
To install the "bleeding edge" version from github via pip, call
```

Comment (GenevieveBuckley) [reply tutorials/fundamentals/installation.md]: I don't think "next-release" is very clear (but "bleeding edge" is also not very clear, so if you have any better ideas we should do that)

**Comment by GenevieveBuckley** [comment]
I think some of my comments were more about content, rather than structure, which might not be what you're looking for on this PR.

File: tutorials/fundamentals/installation.md
Comment (melissawm) [reply tutorials/fundamentals/installation.md]: Development version? Unstable version?

File: tutorials/fundamentals/installation.md
Comment (melissawm) [inline tutorials/fundamentals/installation.md]:
```suggestion
## Choose which distribution to install.
```
Going up one level ensures people understand the next sections are options and they should only go with one of them. I'd go as far as using numbering to make that even clearer:

1. From pip
2. From conda-forge
3. From the main branch on GitHub

File: tutorials/fundamentals/installation.md
Comment (melissawm) [inline tutorials/fundamentals/installation.md]:
```suggestion
### Install from the main branch on Github
```

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md]: Yeah I think it would be nice if this was a dropdown selection similar to [this section](https://numpy.org/numpy-tutorials/content/pairing.html#pair-your-notebook-files-ipynb-and-md) in the numpy notebook/jupytext guide:

1. It would limit the amount of code those who want to install the bundled app see at first glance if scrolling through the document
2. It would limit distractions for those who just want to install via pip or via conda - you probably don't want to do all of them and if you do, it doesn't feel like that much effort to click all three
3. It clearly shows that these are three separate options and in general you would pick just one.

Thoughts? I'm not sure how easy this would be to implement given the current theme because I don't remember seeing any drop down/accordion behaviour. Pinging @codemonkey800 - do we have something like this? If not, could we bring over something like the [FAQ dropdown](https://www.napari-hub.org/faq) from the hub?

**Comment by DragaDoncila** [comment]
> I think some of my comments were more about content, rather than structure, which might not be what you're looking for on this PR.

@GenevieveBuckley we may as well do it while we're here. I don't think any of this intersects with the "Absolute Beginners" document (I feel like we would be linking to that document from here), so I think we should make the improvements you suggest to this document.

File: tutorials/fundamentals/installation.md
Comment (melissawm) [reply tutorials/fundamentals/installation.md]: That is a [jupyter-book feature](https://jupyterbook.org/interactive/hiding.html) so might be already doable - although I don't know about the styling using the napari theme.

Comment (codemonkey800) [reply tutorials/fundamentals/installation.md]: The napari tutorials use this feature extensively, so we'll also have support for it in the napari theme 😄 

You can see a demo of it in #179.

**Comment by melissawm** [comment]
When this is merged, can we remove the duplicate installation instructions from the site home page?

**Comment by GenevieveBuckley** [comment]
> When this is merged, can we remove the duplicate installation instructions from the site home page?

Could you link to both places? I agree that we probably don't want duplication, that's harder to maintain.

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md]: this was to include a horizontal line separator between the content headings and the template headings, but I think stuff like that is supposed to be dictated by the theme instead.

**Comment by DragaDoncila** [comment]
@GenevieveBuckley @melissawm I think this is ready for another round of reviews. I've converted it to MyST markdown so we can take advantage of all the new theme options, and I think I've addressed the feedback I've received. I tried to clean up the sections a bit too, and I've put the three different package install options into dropdowns.

I haven't yet managed to build it with the new theme, I'll make sure I do that before we merge, but content wise I'd love another pass from you both.

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md:245]: Did we lose this content on getting help if needed, or has it just been moved elsewhere in the document? I think the link for where to submit an issue is useful information.

**Comment by GenevieveBuckley** [comment]
I don't have additional comments, this seems mostly fine to me. Nothing much new from me.

There are a few places where I have an outstanding comment or two about not wanting to lose information that was included in this document before (like redirecting people to the developer install guide if they need that instead, how to file an issue if things went wrong with the install, etc.)

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md:245]: this has been moved to the bottom under "Next steps".

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md]: @GenevieveBuckley I added an important note above:

```{important}
Note that the bundled app is still
in active development, and may not be very stable. We strongly recommend installing as a Python
package instead.
```

Is this sufficient mention of the restrictions?

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md:8]: Yep, I've added this at the very top in a note:

```{note} 
If you want to contribute back to the napari codebase, you should follow [this guide instead](https://napari.org/developers/contributing.html).
```

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [reply tutorials/fundamentals/installation.md]: Yes, I think so. Might be good to link to the python installation section/instructions in that sentence, too.

File: tutorials/fundamentals/installation.md
Comment (melissawm) [inline tutorials/fundamentals/installation.md:14]: Do we want to make this document executable?

File: tutorials/fundamentals/installation.md
Comment (melissawm) [inline tutorials/fundamentals/installation.md]: I feel like we have redundant information here, since the different installation methods have already been mentioned above. Maybe the "Choose which distribution to install" section could come before the particular installation methods.

**Review by melissawm — COMMENTED** [review]
Thanks @DragaDoncila , I only have one observation about the content, overall looks good (and once this is merged I would suggest cleaning up the index page!)

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md]: Yeahh this one is a bit weird because "Choose which distribution to install" does only refer to installing as package - it's just do you want it from `pip`, `conda` or `github main branch`. I'll change the wording here to reflect this a bit better.

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md:14]: I've made this document executable for now purely so I can use the Jupyter book dropdown feature, and also because I think in general we want all tutorials to be MyST markdown eventually. 
Once we merge the dropdown accordion feature (issue #279) we can change this dropdown to match that styling. Do you think there are any dangers to keeping it executable until then @melissawm?

**Comment by DragaDoncila** [comment]
Ok so @GenevieveBuckley and @melissawm I actually don't need to move this to `napari/napari` since tutorials are in `napari.github.io` anyway. 

I'd love if either one of you could do a final pass on this one. I think I've addressed all the outstanding comments? I've replied to each comment I think.

I'll also add a couple of screenshots here to show what it looks like with the theme:
Screen Shot 2021-11-30 at 11 47 14 AM
Screen Shot 2021-11-30 at 11 47 42 AM

It might also be good to get a review from someone outside the docs working group. @sofroniewn @tlambert03?

**Review by sofroniewn — APPROVED** [review]
LGTM! Thanks for updating!!

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
If you want to contribute code back into napari, you should follow the [development installation instructions in the contributing guide](https://napari.org/developers/contributing.html) instead.
```

I'd kind of like the words "development installation" included in this sentence, otherwise it's a bit hard to see how this page content relates to this note.

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
- [Python 3.7 or higher](https://www.python.org/downloads/)
```

Just for clarity :smile:

Comment (GenevieveBuckley) [reply tutorials/fundamentals/installation.md]: Although, I'm not sure you should link to https://www.python.org/downloads/ if you are later going to try to encourage the use of conda. Maybe the link can be removed?

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
While not strictly required, it is highly recommended to install
```

Personal preference, I'd prefer we didn't have such a mix of emphasis markers so close together. (There is bold text above for **(Highly recommended)**, the "Important" box is also for emphasis, I don't think we need a single word in italics to further emphasize things on top of all that)

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
python -m pip install "napari[all]"
```

Suggestion: we should consider making it a general rule that anytime we have instructions with pip, we show `python -m pip ...`. It reduces the chance that someone unfamiliar with virtual environments will end up pip installing packages into their base environment instead by mistake, etc.

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
python -m pip install "napari[all]" --upgrade
```

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]: This is my favourite docs page on how to use conda environments, maybe it's worthwhile linking to it here? Up to you.
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
python -m pip install "git+https://github.com/napari/napari.git#egg=napari[all]"
```

**Review by GenevieveBuckley — APPROVED** [review]
Looking great, @DragaDoncila 

I've left some minor comments, but I think we can merge this in the next day or so. Good work!

File: tutorials/fundamentals/installation.md
Comment (DragaDoncila) [reply tutorials/fundamentals/installation.md]: Lovely, thanks Gen!

**Comment by DragaDoncila** [comment]
@GenevieveBuckley thanks for that final round of very thoughtful comments! I've updated and I think this is ready now.

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
### Prerequisites for installing napari as a Python package 
```

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
### Prerequisites for installing napari as a bundled app
```

File: tutorials/fundamentals/installation.md
Comment (GenevieveBuckley) [inline tutorials/fundamentals/installation.md]:
```suggestion
## Install as Python package (recommended)
```

**Comment by GenevieveBuckley** [comment]
Looks good!

I've just updated some of the headings to make it clearer which are part of the prerequisites section and which are part of the instructions section, and will merge the PR now.

Thanks for working on this @DragaDoncila

**Pull request merged**
